server/src/test/java/org/elasticsearch/index/query/SimpleQueryStringBuilderTests.java

```diff
@@ -104,6 +104,12 @@ protected SimpleQueryStringBuilder doCreateTestQueryBuilder() {
                 fields.put(STRING_FIELD_NAME_2, 2.0f / randomIntBetween(1, 20));
             }
         }
+        // special handling if query is "now" and no field specified. This hits the "mapped_date" field which leads to the query not being
+        // cacheable and trigger later test failures (see https://github.com/elastic/elasticsearch/issues/35183)
+        if (fieldCount == 0 && result.value().equalsIgnoreCase("now")) {
+            fields.put(STRING_FIELD_NAME_2, 2.0f / randomIntBetween(1, 20));
+        }
```

Comment:
I don't like really like adding so many lines for a relatively rare case, but every other way (e.g. disallowing "now" altogether as a query string) would have been worse IMHO.